bin/steps/clang

```diff
@@ -2,8 +2,8 @@ if [[ ! -d "$CACHE_DIR/clang-$CLANG_VERSION" ]]; then
   cd $CACHE_DIR
   puts-step "Installing clang-$CLANG_VERSION"
   mkdir -p "clang-$CLANG_VERSION"
-  curl http://llvm.org/releases/$CLANG_VERSION/clang+llvm-$CLANG_VERSION-x86_64-linux-gnu-ubuntu-14.04.tar.xz -s -L | xz -d -c | tar x -C clang-$CLANG_VERSION &> /dev/null
+  curl http://llvm.org/releases/$CLANG_VERSION/clang+llvm-$CLANG_VERSION-x86_64-linux-gnu-ubuntu-16.04.tar.xz -s -L | xz -d -c | tar x -C clang-$CLANG_VERSION &> /dev/null
```

Comment:
This breaks compilation in projects that still use the older `cedar-14` stack. Even though `heroku-16` is the new default, old projects will be broken all around the world if this would get merged.

Please find another, preferably dynamic way to fetch either the 14.04 or 16.04 version, depending on the current system.

You will find similar logic in other steps.